agent/src/main/java/io/pyroscope/javaagent/impl/ContinuousProfilingScheduler.java

```diff
@@ -38,14 +39,76 @@ public ContinuousProfilingScheduler(Config config, Exporter exporter, Logger log
 
     @Override
     public void start(Profiler profiler) {
-        Duration firstProfilingDuration;
+        this.logger.log(Logger.Level.DEBUG, "ContinuousProfilingScheduler#start");
+        synchronized (lock) {
+            if (started) {
+                throw new IllegalStateException("already started");
+            }
+            Duration firstProfilingDuration;
+            try {
+                firstProfilingDuration = startFirst(profiler);
+            } catch (Throwable throwable) {
+                stopSchedulerLocked();
+                throw new IllegalStateException(throwable);
+            }
+            this.profiler = profiler;
+            this.executor = Executors.newSingleThreadScheduledExecutor(THREAD_FACTORY);
+            this.job = executor.scheduleAtFixedRate(this::schedulerTick,
+                firstProfilingDuration.toMillis(), config.uploadInterval.toMillis(), TimeUnit.MILLISECONDS);
+            this.started = true;
```

Comment:
perhaps a logging statement here could be helpful

Reply:
added